x-pack/plugins/enterprise_search/public/applications/app_search/components/engines/engines_logic.ts

```diff
@@ -0,0 +1,116 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { kea, MakeLogicType } from 'kea';
+
+import { HttpLogic } from '../../../shared/http';
+
+import { EngineDetails } from '../engine/types';
+
+interface EnginesValues {
+  dataLoading: boolean;
+  engines: EngineDetails[];
+  enginesTotal: number;
+  enginesPage: number;
+  metaEngines: EngineDetails[];
+  metaEnginesTotal: number;
+  metaEnginesPage: number;
+}
+
+interface OnEnginesFetch {
+  engines: EngineDetails[];
+  total: number;
+}
+interface EnginesActions {
+  onEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onMetaEnginesFetch({ engines, total }: OnEnginesFetch): OnEnginesFetch;
+  onEnginesPagination(page: number): { page: number };
+  onMetaEnginesPagination(page: number): { page: number };
+  fetchEngines(): void;
+  fetchMetaEngines(): void;
+}
+
+export const EnginesLogic = kea<MakeLogicType<EnginesValues, EnginesActions>>({
```

Comment:
This is not a 1:1 translation from ent-search's `EnginesLogic` b/c the view/table component is actually fairly different from the standalone UI. Quick sparknotes of changes:

- Meta engines is on this view as well, not just source engines
- Flash messages is now a global state, not per-view
- We currently don't have engines filtering in, I'm not sure if this is something Davey eventually wants post-MVP

We're also missing a delete engine/create engine listener which I assume we'll add at some point (I guess create engine could use a different logic file actually on second thought?)